### PR TITLE
fix: dateDiff minutes returns wrong value across hour boundary

### LINKED ISSUE #11657: The `dateDiff` function produces a wrong interval in minutes
T27785 - The dateDiff function produces a wrong interval in minutes  
https://surveyjs.answerdesk.io/internal/ticket/details/T27785

### PR DESCRIPTION
**Pull request by JaneSjs**
## Summary
- Fix `dateDiff(..., ''hours''|''minutes''|''seconds'')` to use a millisecond delta (same approach as v2/master)
- The old calendar-day adjustment incorrectly decremented days when crossing an hour, so `dateDiff(''2026-07-31T00:58'', ''2026-07-31T01:05'', ''minutes'')` returned `-1493` instead of `7`
- Add regression tests for hours, minutes (including hour-boundary), and seconds

## Test plan
- [ ] `dateDiff({start}, {end}, ''minutes'')` with start `2026-07-31T00:58` and end `2026-07-31T01:05` returns `7`
- [ ] Existing Bug#10177 case (`13:00` → `13:05`) still returns `5`
- [ ] Hours/seconds cases in `expressionParserTest.ts` pass

### CODE REVIEW

File: packages/survey-core/src/functionsfactory.ts [hunk 1]
```diff
@@ -263,23 +263,15 @@ FunctionFactory.Instance.register("age", age);
...
-    }
-    let hours = days * 24 + date2.getHours() - date1.getHours();
-    if (type === "hours") return hours;
-    if (date2.getMinutes() < date1.getMinutes()) {
-      hours -= 1;
-    }
-    return hours * 60 + date2.getMinutes() - date1.getMinutes();
+  if (type === "hours" || type === "minutes" || type === "seconds") {
+    const date1: any = createDate("function-dateDiffMonths", params[0]);
+    const date2: any = createDate("function-dateDiffMonths", params[1]);
```
Comment (MarchTokenda) [inline packages/survey-core/src/functionsfactory.ts]:
```suggestion
    const date2: any = createDate("function-dateDiff", params[1]);
```

File: packages/survey-core/src/functionsfactory.ts [hunk 2]
```diff
@@ -263,23 +263,15 @@ FunctionFactory.Instance.register("age", age);
...
-      days -= 1;
-    }
-    let hours = days * 24 + date2.getHours() - date1.getHours();
-    if (type === "hours") return hours;
-    if (date2.getMinutes() < date1.getMinutes()) {
-      hours -= 1;
-    }
-    return hours * 60 + date2.getMinutes() - date1.getMinutes();
+  if (type === "hours" || type === "minutes" || type === "seconds") {
+    const date1: any = createDate("function-dateDiffMonths", params[0]);
```
Comment (MarchTokenda) [inline packages/survey-core/src/functionsfactory.ts]:
```suggestion
    const date1: any = createDate("function-dateDiff", params[0]);
```

**Review by MarchTokenda — APPROVED** [review]
